### PR TITLE
Fix segmentation results processing

### PR DESCRIPTION
**Pull request by druzhkov-paul**
Build 21388 ✔️ 
Tests 21452 ✔️

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/entities/shapes/rectangle.py [hunk 1]
```diff
@@ -67,8 +68,14 @@ def __init__(
             modification_date=modification_date,
         )
 
+        is_valid = True
         for (x, y) in [(x1, y1), (x2, y2)]:
-            self._validate_coordinates(x, y)
+            is_valid = is_valid and self._validate_coordinates(x, y)
+        if not is_valid:
+            warnings.warn(
+                f"Invalid {type(self).__name__} : x1={x1}, y1={y1}, x2={x2}, y2={y2}",
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/entities/shapes/rectangle.py]: Please suppress warnings in tests OR change warning message to start from `f"{type(self).__name__} coordinates`

**Comment by druzhkov-paul** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/utils/segmentation_utils.py [hunk 2]
```diff
@@ -184,27 +230,34 @@ def create_annotation_from_segmentation_map(
         # Contour retrieval mode CCOMP (Connected components) creates a two-level
         # hierarchy of contours
         contours, hierarchies = cv2.findContours(
-            label_index_map, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE
+            label_index_map, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_NONE
```
Comment (evgeny-izutov) [inline ote_sdk/ote_sdk/utils/segmentation_utils.py:233]: Do we really need to disable the approximation? Seems it doesn't break the logic of subcontour selections but speeds up inference.

File: ote_sdk/ote_sdk/utils/segmentation_utils.py [hunk 3]
```diff
@@ -142,6 +143,51 @@ def create_hard_prediction_from_soft_prediction(
...
 
+Contour = List[Tuple[float, float]]
+
+
+def get_subcontours(contour: Contour) -> List[Contour]:
+    """
+    Splits contour into subcontours that do not have self intersections.
+    """
+
+    ContourInternal = List[Optional[Tuple[float, float]]]
```
Comment (evgeny-izutov) [inline ote_sdk/ote_sdk/utils/segmentation_utils.py:154]: Why do you need to convert it to float and then compare instead of comparing int values?

**Comment by Ilya-Krylov** [comment]
@druzhkov-paul,  Will not it be too difficult for you to add several tests for `get_subcontours` with maybe typical invalid contours and a test that would fail without these changes if the test existed before.

File: ote_sdk/ote_sdk/utils/segmentation_utils.py (hunk 3 above)
Comment (druzhkov-paul) [reply ote_sdk/ote_sdk/utils/segmentation_utils.py:154]: This is just type annotation. It has no effect on the commands that are executed at runtime, i.e. no conversion is happening. It just says mypy then float and its subclasses are acceptable here. And `int` is a subclass of `float`. [PEP-484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower)

**Review by evgeny-izutov — APPROVED** [review]

**Pull request merged**